### ISSUE TITLE
Is it possible to pipe content to shellcheck directly?

### ISSUE THREAD

**Issue by klauern** [koalaman/shellcheck#982]
#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

Can you use `shellcheck` to pipe content to it instead of passing a filename?    I'm not a bash expert on this, so maybe my redirect-fu is lacking.  I've tried the following:

```sh
$ shellcheck < $(cat getMOSPatch.sh)
bash: $(cat getMOSPatch.sh): ambiguous redirect

$ shellcheck < getMOSPatch.sh
No files specified.

Usage: shellcheck [OPTIONS...] FILES...
  -e CODE1,CODE2..  --exclude=CODE1,CODE2..  exclude types of warnings
  -f FORMAT         --format=FORMAT          output format
  -s SHELLNAME      --shell=SHELLNAME        Specify dialect (bash,sh,ksh)
  -V                --version                Print version 

$ cat getMOSPatch.sh | shellcheck
No files specified.

Usage: shellcheck [OPTIONS...] FILES...
  -e CODE1,CODE2..  --exclude=CODE1,CODE2..  exclude types of warnings
  -f FORMAT         --format=FORMAT          output format
  -s SHELLNAME      --shell=SHELLNAME        Specify dialect (bash,sh,ksh)
  -V                --version                Print version information

$ cat getMOSPatch.sh > shellcheck
# outputs nothing
```

**Comment by Pawamoy** [comment]
Some of your commands are not semantically valid.
If you want to use the output of a command as a temporary file for input, you should use the following syntax:

```
shellcheck <(cat getMOSPatch.sh)  # note the absence of $
```

Also, `cat getMOSPatch.sh > shellcheck` will only write output of the cat command to a **file** named shellcheck. Il will not pipe it to the **command** named shellcheck.

These are just syntax observations, not ways to achieve what you would like to do with shellcheck.

**Comment by mineo** [comment]
`cat getMOSPatch.sh | shellcheck` works if you change it to `cat getMOSPatch.sh | shellcheck -`.

**Comment by koalaman** [comment]
Specifying `-` as a filename is the canonical way to get programs to read from stdin and shellcheck supports it. Process substitution and `/device/stdin` works too

**Comment by klauern** [comment]
all,

Thank you for the tips and pointers.  It was indeed my lack of bash redirect knowledge.  As a parting gift, I did find an outstanding cheatsheet on this that I'm going to pore over: http://www.catonmat.net/blog/bash-one-liners-explained-part-three/ 

and the cheatsheet: http://www.catonmat.net/download/bash-redirections-cheat-sheet.pdf
